src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,187 @@
+
+    final static double kP = Preferences.getInstance().getDouble(RobotMap.Preferences.K_P, 1);
+    final static double kI = Preferences.getInstance().getDouble(RobotMap.Preferences.K_I, 1);
+    final static double kD = Preferences.getInstance().getDouble(RobotMap.Preferences.K_D, 1);
+    final static double kF = Preferences.getInstance().getDouble(RobotMap.Preferences.K_F, 1);
+
+    //enumerator that will be pulled from for the GoToHeight Command
+    public enum LiftHeight {
+        //will be changed
+        kFloor(0), kSwitch(1), kPortal(2), kIntake(3), kScaleLow(4), kScaleHigh(5), kClimb(6), kMaxHeight(7);
+
+        @SuppressWarnings("MemberName")
+        private double value;
+
+        //
+        LiftHeight(double value) {
+            this.value = value;
+        }
+
+        public double get() {
+            return value;
+        }
+    }
+
+    public Lift() {
+        super("Lift", kP, kI, kD, kF);
+        //creates a child for the encoders
+        addChild(liftEncoder);
+        addChild(topLimitSwitch);
+        addChild(bottomLimitSwitch);
+
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_P, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_I, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_D, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_F, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.LIFT_UP_DOWN_INCRIMENT, 1);
+    }
+
+    protected double returnPIDInput() {
+        return liftEncoder.getDistance();
+    }
+
+    protected void usePIDOutput(double output) {
+        if (topLimitSwitch.get() == true && output > 0) {
+            setSetpoint(LiftHeight.kMaxHeight.get());
+            return;
+        }
+        if (bottomLimitSwitch.get() == true && output < 0) {
+            setSetpoint(LiftHeight.kFloor.get());
+            return;
+        }
+        liftDrive.set(output);
+    }
+
+    //gear changes
+    public void shiftToHighGear() {
+        liftTransmission.set(Value.kForward);
+    }
+
+    public void shiftToLowGear() {
+        liftTransmission.set(Value.kReverse);
+    }
+
+    //does not do anything
+    public void initDefaultCommand() {
+    }
+
+    public Command GoToHeight(LiftHeight height, boolean isHighGear) {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                if (isHighGear == true) {
+                    shiftToHighGear();
+                } else
+                    shiftToLowGear();
+
+                setSetpoint(height.get());
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        };
+    }
+
+    public Command GoUp() {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                setSetpoint(getSetpoint()
```

Comment:
This should be in execute so that it will continue to change the setpoint while the button is held.